File: src/main/java/xyz/baddeveloper/lwsl/packet/Packet.java
package xyz.baddeveloper.lwsl.packet;

import org.json.JSONObject;

public class Packet {

    private JSONObject object;

    public Packet(){}

    public Packet(JSONObject object) { this.object = object; }

    public JSONObject getObject() { return object; }

    public void setObject(JSONObject object) { this.object = object; }
}

File: src/main/java/xyz/baddeveloper/lwsl/packet/Packet.kt
/*
 * Copyright (c) Creepinson
 */
package xyz.baddeveloper.lwsl.packet

import org.json.JSONObject

open class Packet @JvmOverloads constructor(var jsonObject: JSONObject = JSONObject()) {
    fun isPacket(packet: Class<out Packet?>): Boolean {
        return isPacket(packet.simpleName.replace("Packet", ""))
    }

    fun isPacket(packetName: String): Boolean {
        return jsonObject.has("packetName") && jsonObject.getString("packetName") == packetName
    }

    var name: String?
        get() = jsonObject.getString("packetName")
        set(name) {
            jsonObject.put("packetName", name)
        }

}